File: app/src/main/java/mx/dev/franco/automusictagfixer/ui/faq/QuestionsFragment.java
package mx.dev.franco.automusictagfixer.ui.faq;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import androidx.activity.OnBackPressedCallback;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import java.util.List;
import mx.dev.franco.automusictagfixer.R;
import mx.dev.franco.automusictagfixer.ui.BaseFragment;
import mx.dev.franco.automusictagfixer.ui.MainActivity;

public class QuestionsFragment extends BaseFragment implements FaqAdapter.OnItemClick{

    private List<QuestionItem> mQuestionItems;
    private FaqAdapter mFaqAdapter;
    private RecyclerView mRecyclerView;

    public QuestionsFragment(){}

    public static QuestionsFragment newInstance(){
        return new QuestionsFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        return inflater.inflate(R.layout.fragment_questions, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        mQuestionItems = new ArrayList<>();
        String[] questions = getResources().getStringArray(R.array.questions);

        String[] answers = getResources().getStringArray(R.array.answers);

        mRecyclerView = view.findViewById(R.id.questions_recyclerview);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setItemViewCacheSize(10);
        mFaqAdapter = new FaqAdapter(mQuestionItems, this);
        mRecyclerView.setAdapter(mFaqAdapter);

        for(int y = 0 ; y < questions.length ; y++){
            QuestionItem questionItem = new QuestionItem();
            questionItem.setQuestion(questions[y]);
            questionItem.setAnswer(answers[y]);
            mQuestionItems.add(questionItem);
            mFaqAdapter.notifyItemInserted(mQuestionItems.size()-1);
        }

        requireActivity().getOnBackPressedDispatcher().addCallback(new OnBackPressedCallback(true) {
            @Override
            public void handleOnBackPressed() {
                getActivity().getSupportFragmentManager().popBackStack();
            }
        });
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        //((MainActivity)getActivity()).toggle.syncState();
        ((MainActivity)getActivity()).mActionBar.setTitle(getString(R.string.faq));
        ((MainActivity)getActivity()).mStartTaskFab.hide();
        //mToolbar.setNavigationOnClickListener(v -> getActivity().onBackPressed());

        //Get action bar from toolbar
        //((MainActivity)getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);

    }

    @Override
    public void onItemClick(int position, View view) {
        mFaqAdapter.setActivePosition(position);
    }

    @Override
    public void onDestroy(){
        super.onDestroy();
        mQuestionItems.clear();
        mQuestionItems = null;
        mFaqAdapter = null;
        mRecyclerView = null;
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        menu.clear();
    }


    @Override
    public Animation onCreateAnimation(int transit, boolean enter, int nextAnim) {
        Animation animation = super.onCreateAnimation(transit, enter, nextAnim);

        if (animation == null && nextAnim != 0) {
            animation = AnimationUtils.loadAnimation(getActivity(), nextAnim);
        }

        if (animation != null && getView() != null)
            getView().setLayerType(View.LAYER_TYPE_HARDWARE, null);

        return animation;
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/ui/faq/QuestionsFragment.kt
package mx.dev.franco.automusictagfixer.ui.faq

import android.os.Bundle
import android.view.*
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import mx.dev.franco.automusictagfixer.R
import mx.dev.franco.automusictagfixer.ui.BaseFragment
import mx.dev.franco.automusictagfixer.ui.faq.FaqAdapter.OnItemClick
import mx.dev.franco.automusictagfixer.ui.main.MainActivity
import java.util.*

class QuestionsFragment : BaseFragment(), OnItemClick {
    private var mQuestionItems: MutableList<QuestionItem>? = null
    private var mFaqAdapter: FaqAdapter? = null
    private var mRecyclerView: RecyclerView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.fragment_questions, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        mQuestionItems = ArrayList()
        val questions = resources.getStringArray(R.array.questions)
        val answers = resources.getStringArray(R.array.answers)
        mRecyclerView = view.findViewById(R.id.questions_recyclerview)
        mRecyclerView!!.layoutManager = LinearLayoutManager(activity)
        mRecyclerView!!.setHasFixedSize(true)
        mRecyclerView!!.setItemViewCacheSize(10)
        mFaqAdapter = FaqAdapter(mQuestionItems, this)
        mRecyclerView!!.adapter = mFaqAdapter
        for (y in questions.indices) {
            val questionItem = QuestionItem()
            questionItem.question = questions[y]
            questionItem.answer = answers[y]
            mQuestionItems!!.add(questionItem)
            mFaqAdapter!!.notifyItemInserted(mQuestionItems!!.size - 1)
        }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        (activity as MainActivity?)!!.actionBar!!.title = getString(R.string.faq)
    }

    override fun onItemClick(position: Int, view: View?) {
        mFaqAdapter!!.setActivePosition(position)
    }

    override fun onDestroy() {
        super.onDestroy()
        mQuestionItems!!.clear()
        mQuestionItems = null
        mFaqAdapter = null
        mRecyclerView = null
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        menu.clear()
    }

    override fun onCreateAnimation(transit: Int, enter: Boolean, nextAnim: Int): Animation? {
        var animation = super.onCreateAnimation(transit, enter, nextAnim)
        if (animation == null && nextAnim != 0) {
            animation = AnimationUtils.loadAnimation(activity, nextAnim)
        }
        if (animation != null && view != null) requireView().setLayerType(View.LAYER_TYPE_HARDWARE, null)
        return animation
    }

    companion object {
        @JvmStatic
        fun newInstance(): QuestionsFragment {
            return QuestionsFragment()
        }
    }
}